src/SFA.DAS.EmployerAccounts/Services/CommitmentsV2ApiClient.cs

```diff
@@ -0,0 +1,110 @@
+        private readonly ILogger<CommitmentsV2ApiClient> _logger;        
+
+        public CommitmentsV2ApiClient(HttpClient httpClient, CommitmentsApiV2ClientConfiguration config, ILogger<CommitmentsV2ApiClient> logger) : base(httpClient)
+        {
+            _httpClient = httpClient;
+            _config = config;            
+            _logger = logger;
+        }
+
+        public async Task<GetApprenticeshipResponse> GetApprenticeship(long apprenticeshipId)
+        {
+            await AddAuthenticationHeader();
+
+            var url = $"{BaseUrl()}api/apprenticeships/{apprenticeshipId}";
+            _logger.LogInformation($"Getting GetApprenticeship {url}");
+            var response = JsonConvert.DeserializeObject<GetApprenticeshipResponse>(await GetAsync(url));
+            return response;
+        }
+
+        public async Task<GetApprenticeshipsResponse> GetApprenticeships(GetApprenticeshipsRequest request)
+        {
+            await AddAuthenticationHeader();
+
+            var url = $"{BaseUrl()}api/apprenticeships/?accountId={request.AccountId}&reverseSort={request.ReverseSort}{request.SortField}{request.SortField}{request.SearchTerm}";
+            _logger.LogInformation($"Getting GetApprenticeships {url}");
+            var response = JsonConvert.DeserializeObject<GetApprenticeshipsResponse>(await GetAsync(url));
+            return response;
+        }
+
+        public async Task<GetCohortsResponse> GetCohorts(GetCohortsRequest request)
+        {
+            await AddAuthenticationHeader();
+
+            var url = $"{BaseUrl()}api/cohorts/{request}";
+            _logger.LogInformation($"Getting GetCohorts {url}");
+            var response = JsonConvert.DeserializeObject<GetCohortsResponse>(await GetAsync(url));
+            return response;
+        }
+
+        public async Task<GetDraftApprenticeshipsResponse> GetDraftApprenticeships(long cohortId)
+        {
+            await AddAuthenticationHeader();
+
+            var url = $"{BaseUrl()}api/cohorts/{cohortId}/draft-apprenticeships";
+            _logger.LogInformation($"Getting GetDraftApprenticeships {url}");
+            var response = JsonConvert.DeserializeObject<GetDraftApprenticeshipsResponse>(await GetAsync(url));
+            return response;
+        }
+
+        public async Task<GetApprenticeshipStatusSummaryResponse> GetEmployerAccountSummary(long accountId)
+        {
+            await AddAuthenticationHeader();
+
+            var url = $"{BaseUrl()}api/accounts/{accountId}/summary";
+            _logger.LogInformation($"Getting GetEmployerAccountSummary {url}");
+            var response = JsonConvert.DeserializeObject<GetApprenticeshipStatusSummaryResponse>(await GetAsync(url));
+            return response;
+        }
+
+        public async Task<GetTransferRequestSummaryResponse> GetTransferRequests(long accountId)
+        {
+            await AddAuthenticationHeader();
```

Comment:
AddAuthenticationHeader should be done on the RequestMessage rather than directly on the HttpClient. Here is an example:
https://github.com/SkillsFundingAgency/das-apim-endpoints/blob/6e6c912e2d6e858f78426e206ecf741913470721/src/SFA.DAS.SharedOuterApi/Infrastructure/InternalApiClient.cs#L24

Reply:
done